packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/InternalMutableRealm.kt

```diff
@@ -76,8 +75,14 @@ internal interface InternalMutableRealm : MutableRealm {
     }
 
     override fun deleteAll() {
-        for (schemaClass: KClass<out BaseRealmObject> in configuration.schema) {
-            delete(schemaClass)
+        schema().let { schema: RealmSchema ->
+            for (schemaClass: KClass<out BaseRealmObject> in configuration.schema) {
+                // Deleting asymmetric objects are not supported.
+                val name = schemaClass.realmObjectCompanionOrNull()!!.io_realm_kotlin_schema().name
```

Comment:
`realmObjectCompanionOrThrow` requires a generic argument which we cannot look up here, and we cannot use `schema().classes.filter { it.kind != RealmClassKind.ASYMMETRIC}` since the `delete` method takes a `KClass` not the `RealmClass` coming out of the filter. I don't think we have an API that maps from RealmClass to KClass?

Reply:
Hmm. We have the _online_ mapping between class names and KClasses in 
https://github.com/realm/realm-kotlin/blob/main/packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/schema/CachedClassKeyMap.kt#L47, which you could grab from `LiveRealmReference.schemaMetadata[className]`. Don't know if that simplifies the code though 🤪 .... and I guess we are also not on the hot path here so maybe fine to just leave it.

Reply:
Ah yeah, I did remember we had _something_ :D 

Yes, that will make the code a lot nicer to read.